mobile/lib/features/wallet/create_invoice_screen.dart

```diff
@@ -27,78 +27,140 @@ class _CreateInvoiceScreenState extends State<CreateInvoiceScreen> {
   Amount? amount;
   final TextEditingController _amountController = TextEditingController();
 
+  final _formKey = GlobalKey<FormState>();
+  bool showValidationHint = false;
+
+  final WalletService walletService = const WalletService();
+  final ChannelConstraintsService channelConstraintsService = const ChannelConstraintsService();
+
+  @override
+  void dispose() {
+    _amountController.dispose();
+    super.dispose();
+  }
+
   @override
   Widget build(BuildContext context) {
-    WalletInfo info = context.watch<WalletChangeNotifier>().walletInfo;
-    log("Refresh receive screen: ${info.balances.onChain}");
+    int channelCapacity = channelConstraintsService.getLightningChannelCapacity();
+    int totalMaxAmount = (channelCapacity / 2).floor();
```

Comment:
```suggestion
    int totalMaxAmount = channelConstraintsService.getTotalMaxAmount();
```

I'd suggest instead of doing math in widget, abstract it away into the same channel contstraint service